.github/workflows/ci.yml

```diff
@@ -182,15 +208,32 @@ jobs:
             compiler: clang-13
           - os: macos-12
             compiler: clang-14
+          - os: macos-12
+            compiler: clang-15
+          - os: macos-12
+            compiler: clang-17
 
     runs-on: ${{ matrix.os }}
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
       - name: Install static analyzers
         if: matrix.os == 'ubuntu-22.04'
-        run: sudo apt-get install clang-tidy-14 cppcheck -y -q
+        run: sudo apt-get install cppcheck -y -q
+
+      - name: Install LLVM and Clang
+        if: matrix.os == 'ubuntu-22.04'
+        uses: KyleMayes/install-llvm-action@v1
+        with:
+          version: "17.0"
+      
+      - name: Fix clang installations
+        if: matrix.os == 'ubuntu-22.04'
+        run: |
+          cp ./llvm/bin/clang++ ./llvm/bin/clang++-17
+          cp ./llvm/bin/clang-format ./llvm/bin/clang-format-17
+          cp ./llvm/bin/clang-tidy ./llvm/bin/clang-tidy-17
```

Comment:
Wouldn't these only be necessary for the ubuntu-22.04, clang-17 run?

I'm unsure how the matrix works, but would suspect it to run these steps per compiler with the current setup.

Reply:
The gcc ones are also running clang-tidy

Reply:
But I agree this is not really optimal